src/Webapi/Webapi__Dom/Webapi__Dom__Document.re

```diff
@@ -1,17 +1,13 @@
 module Impl = (T: {type t;}) => {
   external asDocument : T.t => Dom.document = "%identity";
 
-  let asHtmlDocument: T.t => Js.null(Dom.htmlDocument) = [%bs.raw
-    {|
-    function (document) {
-      return document.doctype.name === "html" ?  document : null;
+  let asHtmlDocument: T.t => option(Dom.htmlDocument) = [%raw {|
+    function(document) {
+      if (document instanceof HTMLDocument) return document;
```

Comment:
@yawaramin I forgot to mention, and check for myself, but this has the same problem as `instanceof HTMLElement`. Now that I've merged my upgrade to the new release, I've hit this bug.

Perhaps it would be better to create a re-usable way to find the instance so you're not adding a bunch of `null` checks here as well 🤔

Reply:
Whoops, thanks again. Fixed and published 0.16.2. For the moment I'm not yet sure how best to generalize the check, so just brute-forcing it for now. Would be really cool if everybody supported optional coalescing already!

Reply:
I don't have to support IE11 so I could 😂

I'll work on a PR soon that combines them. It shouldn't be hard.